sdcm/nemesis.py

```diff
@@ -3820,6 +3820,108 @@ def _shrink_cluster(self, rack=None):
         self.log.info("Cluster shrink finished. Current number of nodes %s", num_of_nodes)
         InfoEvent(message=f'Cluster shrink finished. Current number of nodes {num_of_nodes}').publish()
 
+    # TODO: add support for the 'LocalFileSystemKeyProviderFactory' and 'KmipKeyProviderFactory' key providers
+    # TODO: add encryption for a table with large partitions?
+
+    def disrupt_enable_disable_table_encryption_aws_kms_provider(self):
+        # TODO: add check for Scylla that it supports the feature
+        if self.cluster.params.get("cluster_backend") != "aws":
+            raise UnsupportedNemesis("This nemesis is supported only on the AWS cluster backend")
+        if not self.cluster.params.get("aws_instance_profile_name_db"):
+            raise UnsupportedNemesis("This nemesis is supported only when an instance profile is used")
+        # TODO: update the 'kms_host' parameter when  dynamic key creation gets implemented.
+        self._enable_disable_table_encryption(additional_scylla_encryption_options={
+            'key_provider': 'KmsKeyProviderFactory', 'kms_host': 'kms_test'})
+
+    def _enable_disable_table_encryption(self, additional_scylla_encryption_options=None):
+        if additional_scylla_encryption_options and additional_scylla_encryption_options.get(
+                'key_provider', 'N/A') == 'KmsKeyProviderFactory':
+            for node in self.cluster.nodes:
+                is_restart_needed = False
+                with node.remote_scylla_yaml() as scylla_yml:
+                    if not scylla_yml.kms_hosts:
+                        scylla_yml.kms_hosts = {}
+                    # TODO: update the 'kms_hosts' field when the dynamic key creation gets implemented.
+                    if 'kms_test' not in scylla_yml.kms_hosts:
+                        scylla_yml.kms_hosts['kms_test'] = {
+                            'master_key': 'alias/kms_encryption_test',
+                            'aws_region': 'us-east-1',
+                            'aws_use_ec2_credentials': True,
+                        }
+                        is_restart_needed = True
+                if is_restart_needed:
+                    node.configure_kms()
+                    node.process_scylla_args('--logger-log-level kms=trace')
```

Comment:
the sct config doesn't work after restart ? (i.e. I would assume to see this only test case configuration, since we'll want it off by default)

Reply:
It is here to be not dependent on the SCT configuration.

Reply:
I'm just saying we don't want this by default, only when we need it for investigation, it's generating way too many logs.

Reply:
Done